src/runtime_src/doc/toc/xrt_native_apis.rst

```diff
@@ -0,0 +1,618 @@
+
+The kernel handle (or object) is created from the device, XCLBIN UUID and the kernel name. 
+
+.. code:: c
+      :number-lines: 35
+           
+           xuid_t xclbin_uuid;
+           xrtXclbinGetUUID(xclbin,xclbin_uuid);
+
+           xrtKernelHandle kernel = xrtPLKernelOpen(device, xclbin_uuid, "kernel_name");
+           ....
+           ....
+           xrtKernelClose(kernel);
+
+
+In the above code example
+ 
+      - The UUID of the XCLBIN is retrieved by the API ``xrtXclbinGetUUID`` 
+      - The kernel is created by the API ``xrtPLKernelOpen``
+      - The kernel is closed by the API ``xrtKernelClose``
+
+**Note**: For the kernel with more than 1 CU, a kernel handle (or object) should represent all the CUs having identical interface connectivity. If all the CUs of the kernel are not having identical connectivity, the specific CU name(s) should be used to obtain a kernel handle (or object) to represent the subset of CUs with identical connectivity. Otherwise XRT will do this selection internally to select a group of CUs and discard the rest of the CUs (discarded CUs are not used during the execution of a kernel).  
+
+As an example, assume a kernel name is foo having 3 CUs foo_1, foo_2, foo_3. The CUs foo_1 and foo_2 are connected to DDR bank 0, but the CU foo_3 is connected to DDR bank 1. 
+
+       - Opening kernel handle for foo_1 and foo_2 (as they have identical interface connection)
+       
+         .. code:: c++
+               :number-lines: 35
+                  
+                    cu_group_1 = xrtPLKernelOpen(device, xclbin_uuid, "foo:{foo_1,foo_2}");     
+   
+       - Opening kernel handle for foo_3
+          
+         .. code:: c++
+               :number-lines: 35
+                  
+                    cu_group_2 = xrtPLKernelOpen(device, xclbin_uuid, "foo:{foo_3}");     
+
+
+
+**C++**: In C++, xrt::kernel object can be created from the constructor of ``xrt::kernel`` class. 
+
+.. code:: c++
+      :number-lines: 35
+          
+           auto xclbin_uuid = device.load_xclbin("kernel.xclbin");
+           auto krnl = xrt::kernel(device, name, xclbin_uuid); 
+      
+Exclusive access of the kernel's CU
+***********************************
+  
+The API ``xrtPLKernelOpen`` opens a kernel's CU in a shared mode so that the CU can be shared with the other processes. In some cases, it is required to open the CU in exclusive mode (for example, when it is required to read/write CU mapped register). Exclusive CU opening fails if the CU is already opened in either shared or exclusive access. 
+
+.. code:: c
+      :number-lines: 39
+     
+           xrtKernelHandle kernel = xrtPLKernelOpenExclusive(device, xclbin_uuid, "name");
+
+**C++**: When the ``xrt::kernel`` constructor is called with an additional boolean argument set as true, it opens CU in exclusive mode and returns the kernel object.    
+
+.. code:: c++
+      :number-lines: 39
+       
+           auto krnl = xrt::kernel(device, name, xclbin_uuid, true); 
```

Comment:
Do not use random boolean in API. This is cryptic API
```suggestion
           auto krnl = xrt::kernel(device, name, xclbin_uuid, xrt::kernel::exclusive); 
```
Typically ``exclusive`` is a ``constexpr`` value of a private tag class.

Reply:
Agree.  This PR however is documentation not coding.  This should be fixed in code.

Reply:
implementation specific. So tracking through CR, nothing is needed for this PR